Marlin/src/gcode/probe/M102.cpp

```diff
@@ -38,20 +38,19 @@
  *   M102 S0       : Disable adjustable Z height.
  *
  * Negative S values are commands:
- *   M102 S-1       : Read sensor information
+ *   M102 S-1       : Read BDsensor version
+ *   M102 S-2       : Read BDsensor information
  *   M102 S-5       : Read raw Calibration data
  *   M102 S-6       : Start Calibration
  */
 void GcodeSuite::M102() {
-  if (parser.seenval('S'))
-    bdl.config_state = parser.value_int();
-  else
-    M102_report();
-}
-
-void GcodeSuite::M102_report(const bool forReplay/*=true*/) {
-  report_heading(forReplay, F("Bed Distance Sensor"));
-  SERIAL_ECHOLNPGM("  M102 S", bdl.config_state);
+  if (parser.seenval('S')) {
+    const int8_t command = parser.value_int();
+    if (command == BDS_READ_MM)
+      SERIAL_ECHOLNPGM("Bed Distance:", bdl.read(), "mm");
+    else
+      bdl.config_state = command;
+  }
 }
```

Comment:
Is there anything to ensure that `M102` cannot be called in the middle of a print job, or that doing so will not cause strange things to occur?